components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -4689,4 +4696,17 @@ public static boolean hasIntersection(String[] arr1, String[] arr2) {
 
         return false;
     }
+
+    private static boolean isLightweightAPIKeyGenerationEnabled() {
+        try {
+            APIManagerConfiguration config = ServiceReferenceHolder.getInstance().getAPIManagerConfigurationService()
+                    .getAPIManagerConfiguration();
+            String lightweightAPIKeyGenerationEnabled = config.getFirstProperty(APIConstants.LIGHTWEIGHT_API_KEY_GENERATION_ENABLED);
```

Comment:
Add a config null check to gracefully handle config  read time errors

Reply:
It cannot get a null value since we have added a default value in the Default.json